.github/actions/pre-commit-zsh/action.yml

```
@@ -0,0 +1,21 @@
+name: "Test the pre-commit shell scripts in zsh"
+description: "Only works when ran on macos"
+
+runs:
+  using: "composite"
+  steps:
+    - name: Install python and set up Poetry
+      uses: bakdata/ci-templates/actions/python-setup-poetry@v1.5.2
+      with:
+        poetry-version: "1.3.2"
+        python-version: "3.10"
+
+    - name: Install dependencies
+      shell: zsh {0}
+      run: poetry install --no-interaction
+
+    - name: Run pre-commit-hooks
+      shell: zsh {0}
+      run: |
+        poetry run pre-commit run mypy --all-files --show-diff-on-failure
+        poetry run pre-commit run gen-docs --all-files --show-diff-on-failure
```

I could also do it like that to avoid needing changes after each new added hook, but it would add redundancy.

```suggestion
        poetry run pre-commit run --all-files --show-diff-on-failure
```